APMrover2/sailboat.cpp

```diff
@@ -167,24 +167,16 @@ void Sailboat::get_pilot_desired_mainsail(float &mainsail_out)
 // update mainsail's desired angle based on wind speed and direction and desired speed (in m/s)
 float Sailboat::update_sail_control(float desired_speed)
 {
-    float throttle_out = rover.control_mode->get_throttle_out(desired_speed);
-
-    if (!sail_enabled()) {
-        return throttle_out;
-    }
-
-    // if we are motoring relax the sail
+    // if we are motoring relax the sail and use throttle
     if (throttle_state_t == Sailboat_Throttle::FORCE_MOTOR) {
         rover.g2.motors.set_mainsail(100.0f);
-        return throttle_out;
+        return rover.control_mode->get_throttle_out(desired_speed);
```

Comment:
Hi, not a big fan of how the mode class is calling Sailboat's update_sail_control which is then calling back into Mode::get_throttle_out().  That's very confusing for the next developer

Reply:
I guess we could have a  bool function to say if we should call get_throttle_out then call it from within calc throttle, like how the heading stuff works, there is on_indirect_route() to decide if we should call calc_heading()